src/Php83/Php83.php

```diff
@@ -82,4 +83,116 @@ public static function mb_str_pad(string $string, int $length, string $pad_strin
                 return mb_substr(str_repeat($pad_string, $leftPaddingLength), 0, $leftPaddingLength, $encoding).$string.mb_substr(str_repeat($pad_string, $rightPaddingLength), 0, $rightPaddingLength, $encoding);
         }
     }
+
+    public static function str_increment(string $string): string
+    {
+        if ('' === $string) {
+            throw new \ValueError('str_increment(): Argument #1 ($string) cannot be empty');
+        }
+
+        if (!\ctype_alnum($string)) {
+            throw new \ValueError('str_increment(): Argument #1 ($string) must be composed only of alphanumeric ASCII characters');
+        }
+
+        if (\is_numeric($string)) {
+            $offset = stripos($string, 'e');
+            if ($offset !== false) {
+                $char = $string[$offset];
+                $char++;
+                $string[$offset] = $char;
+                $string++;
+
+                switch ($string[$offset]) {
+                    case 'f':
+                        $string[$offset] = 'e';
+                        break;
+                    case 'F':
+                        $string[$offset] = 'E';
+                        break;
+                    case 'g':
+                        $string[$offset] = 'f';
+                        break;
+                    case 'G':
+                        $string[$offset] = 'F';
+                        break;
+                }
+
+                return $string;
+            }
+        }
+
+        return ++$string;
+    }
+
+    public static function str_decrement(string $string): string
+    {
+        if ('' === $string) {
+            throw new \ValueError('str_decrement(): Argument #1 ($string) cannot be empty');
+        }
+
+        if (!\ctype_alnum($string)) {
+            throw new \ValueError('str_decrement(): Argument #1 ($string) must be composed only of alphanumeric ASCII characters');
+        }
+
+        if (\preg_match('/^0[aA0]?$|^[aA]$/', $string)) {
```

Comment:
```suggestion
        if (\preg_match('/\A(?:0[aA0]?|[aA])\z/', $string)) {
```

[PHP Manual](https://www.php.net/manual/en/regexp.reference.meta.php) says:

> | Meta-character | Description |
> |----------------|-------------|
> | `$` | assert end of subject or before a terminating newline (or end of line, in multiline mode) |

It is safer to use the pair `\A` and `\z` instead of `^` and `$` in [PCRE regular expressions](https://www.php.net/manual/en/regexp.reference.escape.php).

Reply:
Done thanks for the tips 👍